tests/test_template_classes.py

```diff
@@ -0,0 +1,324 @@
+from conspiracies import (
+    PromptTemplate1,
+    PromptTemplate2,
+    MarkdownPromptTemplate1,
+    MarkdownPromptTemplate2,
+    XMLStylePromptTemplate,
+    StringTriplet,
+)
+from conspiracies.data import load_gold_triplets
+
+
+def get_examples_task_introduction():
+    docs, triplets = load_gold_triplets()
+    examples = (docs[:5], triplets[:5])
+    task_description = "This is a test task description"
+    test_tweet = "@user1: This is a test tweet, I am commenting on something someone else said. @user2 this is not good enough."  # noqa: E501
+    return examples, task_description, test_tweet
+
+
+def test_PromptTemplate1():
```

Comment:
You can generalise these tests to:

```
def test_PromptTemplate(template, target, expected_prompt):
  template_instance = temaplate(task_description=None, examples=None)
  prompt = template.create_prompt(target)
  assert expected_prompt == prompt
```

Then you can use [parametrize](https://docs.pytest.org/en/6.2.x/parametrize.html) to test the code run all the tests at once.

```
@pytest.mark.parametrize("template, target,expected_prompt", [
    (PromptTemplate1, test_tweet, expected_prompt1),
    (PromptTemplate2, test_tweet, expected_prompt2),
])
def test_PromptTemplate(template, target, expected_prompt):
    # ...
```
This makes it very easy to add additional templates and tests for them in the future.

Reply:
I would also split this test in two:

1) test_prompt_template_create_prompt
2) test_prompt_template_parse_prompt

This makes it easy to see when running the tests what part does not work.